src/pages/List/List_Products.js

```diff
@@ -1,90 +1,53 @@
 import React, { useEffect, useRef, useState } from 'react';
-import { Link, Outlet } from 'react-router-dom';
 import { onSnapshot } from 'firebase/firestore';
-
 import { getListFromDB } from '../../lib/api';
-
 import './ListProducts.css';
 import { Redirection } from '../../components/Redirection';
 import { Nav } from '../../components/Nav';
+import ListEmpty from '../../components/ListEmpty/ListEmpty';
+import FormProducts from '../../components/formProducts/FormProducts';
+import Loading from '../../components/loading/loading';
 
-function ListProducts() {
-  const [items, setItems] = useState([]);
-  const [itemName, setItemName] = useState('');
-  const list = useRef({});
+const ListProducts = () => {
+  const [itemsProducts, setItemsProducts] = useState([]);
   const token = useRef(localStorage.getItem('token'));
+  const [loading, setLoading] = useState(true);
 
   useEffect(() => {
     if (token.current) {
       /* Get items */
       const unsubscribe = onSnapshot(getListFromDB(token.current), (doc) => {
-        list.current = doc.data();
-        setItems(doc.data().items);
+        let { items } = doc.data();
+        if (items === undefined) {
```

Comment:
this could also work 
```
       if (items === undefined) {
           setLoading(false);
           setItemsProducts([]);
         }
        setLoading(false);
        setItemsProducts(items);
```

Reply:
fixed!